include/aspect/particle/generator/interface.h

```diff
@@ -126,6 +126,21 @@ namespace aspect
           void
           parse_parameters (ParameterHandler &prm);
 
+          void
+          parse_global_parameters (ParameterHandler &prm);
```

Comment:
I think there is no need for this additional function. If there are parameters that should be read by the interface you can put them into the default implementation of `parse_parameters` and call that function from derived plugins.